.circleci/config.yml

```diff
@@ -3490,6 +3500,27 @@ workflows:
             - slack
             - circleci-repo-readonly-authenticated-github-token
 
+  # Weekly cache warming for upgrade tests.
+  # Runs at 01:00 UTC on Sundays to warm the Foundry RPC cache for the new week's
+  # pinned block number. This ensures faster CI runs throughout the week.
```

Comment:
How much time does warming the cache save? Wondering if this is worth the added code/logic since without this the first job just ends up as the one that warms the cache anyway and is slightly slower